supporter-product-data/cloudformation/cfn.yaml

```diff
@@ -319,6 +352,63 @@ Resources:
                   - states:StartExecution
                 Resource: !Ref SupporterProductData
 
+  SupporterProductDataDeadLetterQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: !Sub dead-letters-supporter-product-data-${Stage}
+      Tags:
+        - Key: Stage
+          Value: !Ref Stage
+
+  SupporterProductDataQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: !Sub supporter-product-data-${Stage}
+      RedrivePolicy:
+        deadLetterTargetArn:
+          Fn::GetAtt:
+            - "SupporterProductDataDeadLetterQueue"
+            - "Arn"
+        maxReceiveCount: 10
+      VisibilityTimeout: 3000
+      Tags:
+        - Key: Stage
+          Value: !Ref Stage
+
+  SQSTrigger:
+    Type: AWS::Lambda::EventSourceMapping
+    Properties:
+      BatchSize: 1
+      Enabled: true
+      EventSourceArn: !GetAtt SupporterProductDataQueue.Arn
+      FunctionName: !Ref ProcessSupporterRatePlanItemLambda
+    DependsOn:
+      - SupporterProductDataQueue
+      - ProcessSupporterRatePlanItemLambda
+
+  UnprocessedSupporterProductDataRecordAlarm:
+    Type: 'AWS::CloudWatch::Alarm'
+    Condition: CreateProdResources
+    Properties:
+      AlarmName: !Sub 'Urgent 9-5 - There was a failure processing a supporter record in the ProcessSupporterRatePlanItemLambda lambda'
+      AlarmDescription: >
+        Check the dead-letters-direct-mail-PROD SQS queue for details of the record which failed and the ProcessSupporterRatePlanItemLambda
+        Cloudwatch logs for details of what went wrong
+      Namespace: AWS/SQS
+      Dimensions:
+        - Name: QueueName
+          Value: !Sub dead-letters-direct-mail-${Stage}
+      MetricName: ApproximateNumberOfMessagesVisible
+      Statistic: Average
+      ComparisonOperator: GreaterThanThreshold
+      Threshold: '1'
+      Period: '60'
+      EvaluationPeriods: '60'
```

Comment:
worth noting that because the dead letter queue doesn't expire, this will go off just once, and never again until the queue is cleared.  Not sure if it's worth also having a threshold 100 or so for "this is really going off and noone paid attention last time"